frontend/libs/studio-components/src/components/StudioToggleableTextfield/StudioTextfieldToggleView/StudioTextfieldToggleView.tsx

```diff
@@ -4,21 +4,29 @@ import { StudioButton, type StudioButtonProps } from '@studio/components';
 import classes from './StudioTextfieldToggleView.module.css';
 import cn from 'classnames';
 
-export type StudioTextfieldToggleViewProps = StudioButtonProps;
+export type StudioTextfieldToggleViewProps = StudioButtonProps & {
+  label?: string;
+};
 
 export const StudioTextfieldToggleView = ({
   onClick,
   children,
+  title,
+  label,
   className: givenClass,
+  icon = <KeyVerticalIcon data-testid='keyIcon' aria-hidden />,
   ...rest
 }: StudioTextfieldToggleViewProps) => {
   const className = cn(classes.button, givenClass);
 
   return (
     <StudioButton className={className} onClick={onClick} {...rest}>
-      <span className={classes.viewModeIconsContainer}>
-        <KeyVerticalIcon data-testid='keyIcon' aria-hidden />
-        {children}
+      <span className={classes.viewModeIconsContainer} title={title}>
+        {icon}
+        <div className={classes.ellipsis}>
+          {label && <div className={classes.label}>{label}</div>}
+          <div className={classes.ellipsis}>{children}</div>
+        </div>
```

Comment:
I utgangspunktet skal man ikke bruke `<div>` inni `<span>` og `<button>`. Det er fordi `<span>` og `<button>` er såkalt [phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2) og kan derfor bare ha innhold som også er i den kategorien. Som regel går det fint (selv designsystemet gjør det noen steder), men det kan forårsake uventede CSS-problemer. Jeg ville brukt `<span>` i stedet for `<div>` her.

Reply:
Then the css will work differently. I changed the css to work with `span`